src/main.ts

```diff
@@ -257,7 +257,9 @@ export class Minifier {
       // Property names cannot start with a number. Generate next possible property name that starts
       // with the first alpha character.
     } else if (chars[0].match(/[0-9]/)) {
-      return (firstAlpha + Array(len).join(firstChar));
+      newName = firstAlpha + Array(len).join(firstChar);
+      this.lastGeneratedPropName = newName;
```

Comment:
Before, `this.lastGeneratedPropName` was being set inside the `renameProperty` after `generateNextPropertyName` returned the new property name. There was a reason why it was moved into the `generateNextPropertyName` method:

```
would it be less error prone to do the assignment into this.lastGeneratedPropName within the method generateNextPropertyName? Then the method would be in control of its invariant, and it would be more obvious how it works.
```

But I'm not so sure that it's less error prone anymore after this, because we have to return in so many different places.